### ISSUE TITLE
SC2282 - variables starting with numbers aren't flagged as errors if done in arithmetic context

### ISSUE THREAD

**Issue by RVC2020** [koalaman/shellcheck#2958]
#### For bugs
- Rule Id: SC2282 
- My shellcheck version (`shellcheck --version` or "online"):  0.10.0
- [x] The rule's wiki page does not already cover this 
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```

#!/bin/bash
((1num=1+2))
2num=$((10+1))

```
#### Here's what shellcheck currently says:
In test.sh line 3:
2num=$((10+1))
^------------^ SC2282 (error): Variable names can't start with numbers, so this is interpreted as a command.

#### Here's what I wanted or expected to see:
1num also be flagged as an error under SC2282,
